### ISSUE TITLE
Atom freezing when modifying stylesheet

### ISSUE THREAD

**Issue by postcasio** [abe33/atom-color-highlight#69]
Since upgrading to 0.168.0, when I save my `styles.less`, Atom freezes for about 30 seconds to a minute. I did a CPU profile and this package appears to cause the problem.

![screen shot 2015-01-08 at 09 57 04](https://cloud.githubusercontent.com/assets/1759837/5660869/f5b5042a-971c-11e4-88ca-155f25fdc0d4.png)

**Comment by abe33** [comment]
Hmm weird, I'm not experiencing that, can you provide the `styles.less` that you have so that I can try it?

**Comment by postcasio** [comment]
https://github.com/postcasio/atom-conf/blob/master/styles.less

Also includes a few other stylesheets in the same repo.

I've uploaded a profile if you want to have a look: http://files.casiotone.org/CPU-20150108T101029.cpuprofile

**Comment by abe33** [comment]
Also it appears this is from the `project-palette-finder` and not this package (both relies on [pigments](https://github.com/abe33/pigments) for color parsing)

**Comment by postcasio** [comment]
Ooops. It's early. Want me to move it?

**Comment by postcasio** [comment]
It also seems to be happening when I save a blank LESS file. :(

**Comment by abe33** [comment]
It doesn't really matter at this point, most pigments issues aren't reported on pigments anyway, and when I update pigments I always have to release new versions of both packages.

**Comment by abe33** [comment]
Wow, so I can't reproduce it in a clone of your conf project. I've tried most commands/actions that triggers refresh and new parsing and I never had a significant freeze like the one you described. 
As most of the time is spent in q internal maybe it's related to a problem between two different versions of q specified by two packages that make it breaks when used in pigments. Maybe it'll be interesting to inspect which versions are actually required by packages in your setup.

Also, I don't have access to the cpuprofile you uploaded (403 forbidden).

**Comment by postcasio** [comment]
Oh, man. Turns out this is my fault. I had disabled "exclude VCS ignored paths" temporarily, so the project scan results were including a whole bunch of files from my vendor directories. Apparently 292 files is too many.

With the files ignored again, it is much better. Although there is still a small pause with only 38 files to process. It might be a good idea to split up the processing into batches to prevent the UI locking up.
